pkg/didcomm/protocol/introduce/service.go

```diff
@@ -253,6 +257,28 @@ func (s *Service) doHandle(msg *service.DIDCommMsg, outbound bool) (*metaData, e
 	}, nil
 }
 
+// InvitationReceived is used to finish the state machine
+// the function should be called by didexchange after receiving an invitation
```

Comment:
How does DID Exchange know to do this?

Reply:
I don't know yet. Probably after receiving an invitation did-exchange should check parent thread id. Somehow it should determine that pthid was provided by the introduce protocol.

Reply:
We will need to avoid a direct coupling.